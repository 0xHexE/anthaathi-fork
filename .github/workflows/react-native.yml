name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
