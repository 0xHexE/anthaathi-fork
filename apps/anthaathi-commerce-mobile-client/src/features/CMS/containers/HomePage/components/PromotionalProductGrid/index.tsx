import React from 'react';
import {useIntl} from 'react-intl';
import {ImageBackground, View} from 'react-native';
import {Button, Text, ThemeBase, useTheme} from 'react-native-paper';
import {HomePageComponentType} from '../../../../types/common';
import {useResponsiveValue} from '../../../../utils/useResponsiveValue';

export interface ProductGridProps {
  name: string;
  heading: string;
  description?: string;
  price: number;
  currency: string;
  image: string;
  buttonTitle: string;
  label: string;
  onPress?: () => void;
}

export interface PromotionalProductGridProps {
  products: ProductGridProps[];
}

function PromotionalProductGrid(props: PromotionalProductGridProps) {
  const theme = useTheme();
  const itemWidth = useResponsiveValue(['80%', '100%', '100%', '100%']);

  return (
    <View
      style={{
        flexDirection: itemWidth === '80%' ? 'column' : 'row',
        flexWrap: itemWidth === '80%' ? 'nowrap' : 'wrap',
        marginVertical: 10,
      }}
      testID="promotionalProductGrid">
      {props.products &&
        props.products.map((product, index) => (
          <View key={index}>
            <ProductGrid
              product={product}
              itemWidth={itemWidth}
              theme={theme}
            />
          </View>
        ))}
    </View>
  );
}

export default PromotionalProductGrid;

const ProductGrid = ({
  product,
  itemWidth,

  theme,
}: {
  product: ProductGridProps;
  itemWidth: string;
  theme: ThemeBase;
}) => {
  const intl = useIntl();
  return (
    <View
      style={{
        alignItems: 'center',
        marginHorizontal: itemWidth === '80%' ? 20 : 10,
      }}>
      <View style={{width: itemWidth, alignItems: 'center'}}>
        <ImageBackground
          source={{
            uri: product.image,
          }}
          style={{
            height: 240,
            width: '99%',
          }}>
          <View
            style={{
              height: '100%',
              width: '100%',
              alignItems: 'flex-end',
              justifyContent: 'flex-start',
            }}>
            <Text
              style={{
                color: theme.colors.background,
                fontSize: 14,
                fontWeight: '200',
                paddingHorizontal: 10,
                paddingVertical: 10,
                marginVertical: 5,
                borderRadius: 1,
                backgroundColor: '#313652',
              }}
              variant="titleSmall">
              {product.label}
            </Text>
            <Text
              style={{
                color: '#000',
                fontSize: 14,
                fontWeight: '200',
                paddingHorizontal: 10,
                paddingVertical: 10,
                marginVertical: 5,
                borderRadius: 1,
                backgroundColor: theme.colors.background,
              }}
              variant="titleSmall">
              {intl.formatNumber(product.price, {
                style: 'currency',
                currency: product.currency,
              })}
            </Text>
          </View>
        </ImageBackground>
        <View style={{alignItems: 'center', marginBottom: 10}}>
          <Text
            style={{
              color: '#000',
              fontSize: 14,
              fontWeight: '200',
              marginVertical: 5,
            }}
            variant="titleSmall">
            {product.heading}
          </Text>
          <Text
            style={{
              color: '#000',
              fontSize: 18,
              fontWeight: '200',
              marginVertical: 5,
            }}
            variant="titleSmall">
            {product.name}
          </Text>

          <Text
            style={{
              color: '#000',
              fontSize: 14,
              fontWeight: '200',
              marginVertical: 5,
            }}
            variant="titleSmall">
            {product.description}
          </Text>

          <Button
            mode="contained"
            style={{
              backgroundColor: '#000000',
              borderRadius: 1,
              marginVertical: 5,
            }}
            onPress={product.onPress}>
            {product.buttonTitle}
          </Button>
        </View>
      </View>
    </View>
  );
};

export const PromotionalProductGridCMSInput = {
  _component: HomePageComponentType.PromotionalProductGrid,
  component: PromotionalProductGrid,
};
