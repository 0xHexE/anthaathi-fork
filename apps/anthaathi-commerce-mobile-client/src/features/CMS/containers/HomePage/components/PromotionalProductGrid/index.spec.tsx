import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import PromotionalProductGrid from './index';

describe('PromotionalProductGrid', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PromotionalProductGrid
            products={[
              {
                name: 'Baby Yellow Pepper',
                image:
                  'https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1',
                price: 12,
                currency: 'USD',
                heading: 'Heading',
                buttonTitle: 'Shop',
                label: 'New',
                description: '100% fresh. Sourced from Netherlands',
              },
              {
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                price: 23,
                currency: 'USD',
                heading: 'Heading',
                buttonTitle: 'Shop',
                label: 'New',
                description: '100% fresh. Sourced from Netherlands',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('promotionalProductGrid')).toBeTruthy();
  });
});
