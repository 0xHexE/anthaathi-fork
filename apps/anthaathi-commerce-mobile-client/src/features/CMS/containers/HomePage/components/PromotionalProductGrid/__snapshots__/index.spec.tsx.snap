// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromotionalProductGrid should render the item 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "marginVertical": 10,
    }
  }
  testID="promotionalProductGrid"
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginHorizontal": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "height": 240,
              "width": "99%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 240,
                  "width": "99%",
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "height": "100%",
                "justifyContent": "flex-start",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "backgroundColor": "#313652",
                    "borderRadius": 1,
                    "color": "rgba(255, 251, 254, 1)",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginVertical": 5,
                    "paddingHorizontal": 10,
                    "paddingVertical": 10,
                  },
                ]
              }
            >
              New
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderRadius": 1,
                    "color": "#000",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginVertical": 5,
                    "paddingHorizontal": 10,
                    "paddingVertical": 10,
                  },
                ]
              }
            >
              $12.00
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#000",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginVertical": 5,
                },
              ]
            }
          >
            Heading
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#000",
                  "fontSize": 18,
                  "fontWeight": "200",
                  "marginVertical": 5,
                },
              ]
            }
          >
            Baby Yellow Pepper
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#000",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginVertical": 5,
                },
              ]
            }
          >
            100% fresh. Sourced from Netherlands
          </Text>
          <View
            collapsable={false}
            style={
              Object {
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "#000000",
                    "borderColor": "transparent",
                    "borderRadius": 1,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "marginVertical": 5,
                    "minWidth": 64,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 1,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Object {
                            "writingDirection": "ltr",
                          },
                          Array [
                            Object {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            Object {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            Object {
                              "color": "rgba(255, 255, 255, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      Shop
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginHorizontal": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "height": 240,
              "width": "99%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 240,
                  "width": "99%",
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "height": "100%",
                "justifyContent": "flex-start",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "backgroundColor": "#313652",
                    "borderRadius": 1,
                    "color": "rgba(255, 251, 254, 1)",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginVertical": 5,
                    "paddingHorizontal": 10,
                    "paddingVertical": 10,
                  },
                ]
              }
            >
              New
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderRadius": 1,
                    "color": "#000",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginVertical": 5,
                    "paddingHorizontal": 10,
                    "paddingVertical": 10,
                  },
                ]
              }
            >
              $23.00
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginBottom": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#000",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginVertical": 5,
                },
              ]
            }
          >
            Heading
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#000",
                  "fontSize": 18,
                  "fontWeight": "200",
                  "marginVertical": 5,
                },
              ]
            }
          >
            Capsicum mixed
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Object {
                  "color": "#000",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginVertical": 5,
                },
              ]
            }
          >
            100% fresh. Sourced from Netherlands
          </Text>
          <View
            collapsable={false}
            style={
              Object {
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "#000000",
                    "borderColor": "transparent",
                    "borderRadius": 1,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "marginVertical": 5,
                    "minWidth": 64,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 1,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Object {
                            "writingDirection": "ltr",
                          },
                          Array [
                            Object {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            Object {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            false,
                            Object {
                              "color": "rgba(255, 255, 255, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      Shop
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
